### PR TITLE
Update 04-upgrading-to-v1.4.md

### PR DESCRIPTION
**Pull request by dataders**
## What are you changing in this pull request and why?

Add link to the adapter upgrade guide

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/guides/migration/versions/04-upgrading-to-v1.4.md [hunk 1]
```diff
@@ -27,7 +27,16 @@ Relevant changes:
 
 ### For maintainers of adapter plugins
 
-Link to GH discussion forthcoming
+> **TL;DR** Not much heavy lifting for this minor version. But FYI there will be more work for `1.5.0`. Be warned!
+
+The high-level changes are:
+- Add support for Python 3.11
+- Rename/Replace deprecated exception functions
+- Consolidate timestamp functions & macros
```
Comment (jtcohen6) [inline website/docs/guides/migration/versions/04-upgrading-to-v1.4.md]: I think this change was actually included in v1.3. It landed in the v1.4 changelog because it was merged during v1.3 RC period

File: website/docs/guides/migration/versions/04-upgrading-to-v1.4.md [hunk 2]
```diff
@@ -27,7 +27,16 @@ Relevant changes:
 
 ### For maintainers of adapter plugins
 
-Link to GH discussion forthcoming
+> **TL;DR** Not much heavy lifting for this minor version. But FYI there will be more work for `1.5.0`. Be warned!
```
Comment (jtcohen6) [inline website/docs/guides/migration/versions/04-upgrading-to-v1.4.md]: Should we take a more proactive tone here? v1.5 will require more work, and here's how we plan to help. Just an idea:
```suggestion
> **TL;DR** Not much heavy lifting for this minor version. We anticipate more work for `1.5.0`. We plan to release betas early & often, and provide guidance on upgrading.
```

**Review by jtcohen6 — DISMISSED** [review]
Thanks @dataders! Two non-blocking comments. Feel free to merge as soon as you've addressed (or not, up to you)

**Comment by mirnawong1** [comment]
hey @dataders - is there anything we should on our end for this?

File: website/docs/guides/migration/versions/04-upgrading-to-v1.4.md [hunk 3]
```diff
@@ -27,7 +27,16 @@ Relevant changes:
 
 ### For maintainers of adapter plugins
 
-Link to GH discussion forthcoming
+> **TL;DR** Not much heavy lifting for this minor version. We anticipate more work for `1.5.0`. We plan to release betas early & often, and provide guidance on upgrading.
+
+The high-level changes are:
+- Add support for Python 3.11
+- Rename/Replace deprecated exception functions
+- Consolidate timestamp functions & macros
```
Comment (dbeatty10) [inline website/docs/guides/migration/versions/04-upgrading-to-v1.4.md]:
```suggestion
```

File: website/docs/guides/migration/versions/04-upgrading-to-v1.4.md [hunk 4]
```diff
@@ -27,7 +27,15 @@ Relevant changes:
...
-Link to GH discussion forthcoming
+> **TL;DR** Not much heavy lifting for this minor version. We anticipate more work for `1.5.0`. We plan to release betas early & often, and provide guidance on upgrading.
+
+The high-level changes are:
+- Add support for Python 3.11
+- Rename/Replace deprecated exception functions
+- Add support for Incremental Predicates (if applicable)
+- Make use of new adapter-zone tests
+
+See [dbt-core/discussions/6624](https://github.com/dbt-labs/dbt-core/discussions/6624) for more detailed in for and to ask any questions you may have!
```
Comment (dbeatty10) [inline website/docs/guides/migration/versions/04-upgrading-to-v1.4.md]:
```suggestion
The high-level changes are:
- Add support for Python 3.11
- Rename/replace deprecated exception functions
- Add support for Incremental Predicates (if applicable)
- Make use of new adapter-zone tests

For more detailed information and to ask any questions, please visit [dbt-core/discussions/6624](https://github.com/dbt-labs/dbt-core/discussions/6624).
```

**Review by dbeatty10 — APPROVED** [review]
@dataders I took the liberty to accept a suggestion from Jerco and to make a couple small edits myself.

I'm going to approve & merge now with the intent that you can completely overwrite it with a new version if you wish!
